### PR TITLE
[boost-python] Support Python2 as a feature, fix build with installed Python2

### PR DESCRIPTION
**Pull request by yurybura**

### CODE REVIEW

**Comment by cenit** [comment]
since it is an automatically generated port file, it's better if you update the generating script rather than the artifact, otherwise you are going to loose the mods in the next update ;)

**Comment by yurybura** [comment]
> since it is an automatically generated port file, it's better if you update the generating script rather than the artifact, otherwise you are going to loose the mods in the next update ;)

You right, but this file was already modified after generation (patch was added manually as I see). Can somebody give me a piece of advice how to modify generation script? Thanks!

**Comment by yurybura** [comment]
I've updated generation script. Now it should generate exact same port.

**Comment by JackBoosY** [comment]
@yurybura Please resolve these conflicts.

Thanks.

**Comment by yurybura** [comment]
@JackBoosY I've resolved conflicts and I've merged some other important fixes.

**Comment by JackBoosY** [comment]
@yurybura Also have conflict now.

**Comment by yurybura** [comment]
@JackBoosY Resolved.

**Comment by JackBoosY** [comment]
@yurybura After the CI system is fixed, I will trigger the CI test.

**Comment by yurybura** [comment]
Dear @JackBoosY can you apply this PR?

**Comment by JackBoosY** [comment]
Sorry,  I forgot this PR.

File: ports/boost/CONTROL [hunk 1]
```diff
@@ -3,7 +3,7 @@ Source: boost
 Version: 1.71.0
```
Comment (JackBoosY) [inline ports/boost/CONTROL]: Please update this version to `1.71.0-1`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/boost-mpi/CONTROL [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Automatically generated by boost-vcpkg-helpers/generate-ports.ps1
 Source: boost-mpi
 Version: 1.71.0-1
```
[the remark this answers is not in the document]
Comment (yurybura) [reply ports/boost-mpi/CONTROL]: Done

**Review by JackBoosY — APPROVED** [review]

**Comment by yurybura** [comment]
@ras0219-msft Please have a look this PR..

**Comment by dan-shaw** [comment]
We will have a review by the end of this week. Ping me otherwise.

**Comment by dan-shaw** [comment]
After some discussion, we don't think `python2` is a feature i.e. you can either have python3 or python2, so it doesn't really make sense as a feature. The solution would be to have two ports: `boost-python` which would either point to `boost-python2` or `boost-python3`. In general, we don't really like supporting 2 ports for one library (only in certain exceptions such as `opencv`). So, I'm closing the PR. Please re-open if I'm missing anything.

**Pull request closed without merging**